internal/hud/resourceview.go

```diff
@@ -186,15 +186,54 @@ func (v *ResourceView) resourceExpandedYAML() rty.Component {
 		return rty.EmptyLayout
 	}
 
+	unresourced, otherK8s := unresourcedK8sItems(yi.K8sResources)
+
 	l := rty.NewConcatLayout(rty.DirHor)
 	l.Add(rty.TextString(strings.Repeat(" ", 2)))
 	rhs := rty.NewConcatLayout(rty.DirVert)
-	rhs.Add(rty.NewStringBuilder().Fg(cLightText).Text("The YAML loaded from the Tiltfile includes these K8s objects:").Build())
-	rhs.Add(rty.TextString(strings.Join(yi.K8sResources, "\n")))
+	if unresourced != "" {
+		rhs.Add(rty.NewStringBuilder().Fg(cLightText).Text("Add Dockerfile info in Tiltfile to explore these K8s entities:").Build())
+		rhs.Add(rty.TextString(unresourced))
+		rhs.Add(rty.NewLine())
+	}
+	if otherK8s != "" {
+		rhs.Add(rty.NewStringBuilder().Fg(cLightText).Text("Other K8s entities found in YAML:").Build())
+		rhs.Add(rty.TextString(otherK8s))
+	}
 	l.AddDynamic(rhs)
 	return l
 }
 
+func unresourcedK8sItems(resources []model.YAMLManifestResource) (string, string) {
+	var unresourced []string
+	var otherK8s []string
+
+	for _, r := range resources {
+		if r.Kind == "Deployment" || r.Kind == "Service" {
```

Comment:
ugh, maybe we really should have done this further up the chain.

It turns out the code that joins deployments and services to a manifest actually doesn't refer to deployment or service at all!

It looks like it first pulls out entities with images (which could also be, e.g., jobs or daemonsets) and then pulls out entities with labels that match those entities ([here](https://github.com/windmilleng/tilt/blob/b1a699887da864bb598f1ebb0d20dd251ea56a8c/internal/tiltfile/tiltfile_state.go#L238)).

We should probably have the tiltfile code also group unresourced things if we want to do this right.